src/gd_gd.c

```diff
@@ -400,8 +400,11 @@ BGD_DECLARE(void *) gdImageGdPtr (gdImagePtr im, int *size)
 	void *rv;
 	gdIOCtx *out = gdNewDynamicCtx (2048, NULL);
 	if (out == NULL) return NULL;
-	_gdImageGd (im, out);
-	rv = gdDPExtractData (out, size);
+	if (_gdImageGd (im, out)) {
```

Comment:
@pierrejoye, `_gdImageGd()` is a void function; checking its return value doesn't make sense, and likely causes a compile failure or warning. I think we need to setup CI (building and running the tests) to catch such issues.

Reply:
Thanks @cmb69!

Damned, misread the signature, sorry :/

Reply:
sorry, my mistake :(
@pierrejoye can you please revert this?

Reply:
Reverted. Not that critical as the GD format is really really for dev/testing only. However open for an improved fix :)

Also, as @cmb69 mentioned, GD/GD2 formats are going to be deprecated and removed for 3.0